pytorch_lightning/callbacks/model_summary.py

```diff
@@ -46,11 +46,11 @@ class ModelSummary(Callback):
         >>> trainer = Trainer(callbacks=[ModelSummary(max_depth=1)])
     """
 
-    def __init__(self, max_depth: Optional[int] = 1):
+    def __init__(self, max_depth: int = 1) -> None:
         self._max_depth: int = max_depth
```

Comment:
reviewer: here in the new callback we don't need to type it None, because it does not mix with the deprecated "mode" argument that the `ModelSummary` has.